### PR TITLE
Pre-Publish Checklist: Add expand 'hidden page' functionality

### LINKED ISSUE #9350: Checklist thumbnail to show more affected pages has no action
## Bug Description

As originally reported in this support topic https://wordpress.org/support/topic/checklists-show-more-button-not-showing-more, there is no action when clicking on the thumbnail to show more affected pages. 

## Expected Behaviour
To show more pages affected by the checklist warning / drive the user to those affected pages.

## Steps to Reproduce

Create a story with **more than 4 page**s without text
Open checklist
Click on the + 5 thumbnail to show more pages 
Note inaction 

Screen Shot 2021-10-11 at 11 58 00 AM

## Additional Context

- Plugin Version: 1.12.0
- WordPress Version:
- Operating System: macOS
- Browser: Chrome

## Feature Description
(Updated Dec 15, 2021) 

We're going to add a toggle to expand the thumbnails to see all as a grid. The toggles will persist (just like the panel accordions) for the duration of the user's session until the toggle disappears (all issues or enough issues are resolved for it to go away). 

Figma link: https://www.figma.com/file/V452QsH7ZJhwmAuHtC0xOj/Pea-Pod-Features?node-id=33%3A4300 

Description from @agingoldseco  from [slack thread](https://xwp.slack.com/archives/C0134J07HV3/p1639591784046700): 
> User can click the +8 button OR "Expand" to open the other pages but also has the flexibility to collapse it if they want
Expand/Collapse + arrow convention can be the same as the new pages carousel one
If the user expands the cards, they can stay expanded for the duration of their session even if they close the checklist
My thinking here is that people with small screens will likely have to close the checklist a lot to address checklist issues cause it's in the way of the canvas

Screen Shot 2021-12-14 at 11 31 48 AM

As far as keyboard interaction goes with the dual 'click +8 button or "expand" to open' - since the "+8 button" is actually not a button let's just have the next focusable element after the last actual button in the initial grid row (that would be the 3rd column) just drop to the 'expand' button since that's a real button and it's redundant to have the same button in the tab order twice.

## Acceptance Criteria 
- toggle added to checklist per the above description and details 
- component added to storybook within card structure
- relevant tests 
- keyboard accessibile

### PR DESCRIPTION
**Pull request by samwhale**
## Context

#9350 Feature request

## Summary

Adds functionality to toggle whether page previews are shown in the checklist.

## To-do

- [x] Add tests somewhere

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

A lot changed here. Every card that has thumbnails on it should be able to be expanded/collapsed. Checking every issue will be tedious 😕 so sorry. The best way to do this is trigger the error on one page and then duplicate that page at least 5 times.

Errors that will display a button *if* there are more than 5 pages with the issue.

|Issue|How to trigger
|--|--|
|Element with link too big|Add an element to a page with a link and make the element very big|
|Element with link too small|Add an element to a page with a link and make the element very small|
|Image element missing alt|Add an image and remove the description under the `Accessibility` panel|
|Image element resolution|Add an image with very low resolution or a very small image from your computer (less than 828px x 1792px|
|Text contrast too low with page background|Add text that has low color contrast with the page background|
|Page too little text|Don't add any text to the page|
|Page too many links|Add more than 3 elements with links to a page|
|Page too much text|Add more than 200 characters to a page|
|Text font size too small|Add text with a font size less than 12px|
|Video too long|Add a video longer than 60 seconds|
|Video no captions|Add a video without captions (most videos don't have captions)|
|Video no description|Add a video and remove the description text under the `Accessibility` panel|
|Video no poster|Remove the poster image for the video|
|Video element resolution too low|Add a video that is less than 852px x 480px|

Also, when optimizing multiple videos that will also show the collapse/expand button in the checklist.

All other issues should not show the collapse/expand button.

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9350

### CODE REVIEW

**Review by BrittanyIRL — APPROVED** [review]
I really like this implementation / lift. works great in firefox, safari, chrome ✅  

I don't think the failing tests have anything to do with this PR but let's see if other PRs getting merged doesn't get all of pipeline passing here before a merge.

File: packages/story-editor/src/components/checklist/checks/shared/test/thumbnailWrapper.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { ThumbnailWrapper } from '..';
+import { MAX_THUMBNAILS_DISPLAYED } from '../../../../checklistCard';
+
+// eslint-disable-next-line react/prop-types
+const Wrapper = ({ children }) => (
+  <ThemeProvider theme={dsTheme}>{children}</ThemeProvider>
+);
+
+describe('ThumbnailWrapper', () => {
+  afterEach(cleanup);
```
Comment (maxyinger) [inline packages/story-editor/src/components/checklist/checks/shared/test/thumbnailWrapper.js]: I wonder if there's like a root level `afterEach` we could put that in. Is that a thing? or are they always within the scope of the describe statement?

**Review by maxyinger — APPROVED** [review]
Looks great!

**Comment by samwhale** [comment]
TODO for samuel:
- [x] Add pointer for extra button
- [x] Remove visually hidden text that's being skipped
- [x] Check on `cleanup` code, why it would be necessary (probably not necessary)

**Pull request merged**